### ISSUE TITLE
Update the code sending commands back to Kakoune to directly connect to the UNIX socket

### ISSUE THREAD

**Issue by phaazon** [phaazon/kak-tree-sitter#227]
The initial implementation I tried _many commits ago_ tried to use the socket path directly, but I was constantly failing. The reason was that I was missing the protocol, which, back then, was not documented / hard to find.

Looks like [it’s now doable](https://github.com/tomKPZ/pykak/blob/main/kak_socket.py#L39-L57). Context [here](https://github.com/mawww/kakoune/pull/5099#issuecomment-1941876853).

**Comment by mawww** [comment]
Note that the protocol used on the socket is ad-hoc and might change at any point.

**Comment by phaazon** [comment]
Yeah, it should probably be documented somewhere? I don’t see myself implementing this issue right away; I think we should benchmark to see whether using `kak -p` starts being a bottleneck, vs. directly writing to the socket.

**Comment by hadronized** [comment]
I’m removing `v0.6` as milestone, as this is just purely an optimization and somewhat maybe completely unneeded. We’ll see.
